JavaFXUserInterface/src/main/java/controllers/AssetsController.java

```diff
@@ -168,18 +176,33 @@ private void sortingSetUp() {
                 switch (newValue) {
                     case SORT_DEFAULT:
                         FXCollections.sort(assets, Comparator.comparingInt(Item::getId));
+                        if (!search.getText().trim().isEmpty()) {
```

Comment:
Having the conditional statement every time for every switch case is a little redundant. Maybe you could use the setAssetToDisplay() method to establish a local assets list for the sort asset button to use. You would establish that local list before the switch statement. This way you could remove all the added conditional statements and keep the initial switch case with the local list of assets to display instead of the class member assets

Reply:
I tried this solution at first. However, I was not able to make it work. For some reason, the list of assets would not get generated again even though the code did execute generateThumbnails() properly.

Reply:
The solution we talked about did not work, I tried some other things without any success. I think the way it is now is the better option so far.

Reply:
That is fine then. Leave it this way for now. WE will see when we implement more filters and sorting options.